--- ../imagefs/etc/samba/smb.conf	2016-08-11 18:22:03.859716404 +0100
+++ ../imagefs/etc/samba/smb.conf	2016-08-11 18:22:03.859716404 +0100
@@ -185,6 +185,13 @@
 # public shares, not just authenticated ones
    usershare allow guests = yes
 
+
+# grahams - will allow some symlink following, globally disable here
+unix extensions = no
+
+# Work better inside corporate network
+map untrusted to domain = yes
+
 #======================= Share Definitions =======================
 
 # Un-comment the following (and tweak the other settings below to suit)
@@ -260,0 +268,13 @@
+
+[xilinx]
+path = /home/xilinx
+browsable =yes
+writable = yes
+guest ok = yes
+read only = no
+valid users = samba xilinx
+follow symlinks = yes
+wide links = yes
+create mask = 664
+directory mask = 755
+force user = xilinx
