From c96e1cac59722005484ff5051b03fcc43efd246f Mon Sep 17 00:00:00 2001
From: Nagaraju Mekala <nmekala@xilix.com>
Date: Thu, 28 Jan 2016 14:07:34 +0530
Subject: [PATCH 09/31] fixing the MAX_OPCODES to correct value

Signed-off-by: Nagaraju Mekala <nagaraju.mekala@xilinx.com>
---
 opcodes/microblaze-opc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/opcodes/microblaze-opc.h b/opcodes/microblaze-opc.h
index 292e3a4..4e69f76 100644
--- a/opcodes/microblaze-opc.h
+++ b/opcodes/microblaze-opc.h
@@ -102,7 +102,7 @@
 #define DELAY_SLOT 1
 #define NO_DELAY_SLOT 0
 
-#define MAX_OPCODES 291
+#define MAX_OPCODES 299
 
 struct op_code_struct
 {
-- 
2.7.4

