--- /etc/environment	2022-05-05 15:19:02.001848967 -0700
+++ environment	2022-05-05 16:06:59.618347901 -0700
@@ -1 +1,5 @@
-PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
+PATH="/opt/sigrok/bin:/opt/microblazeel-xilinx-elf/bin:/opt/python3.10/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games"
+LC_ALL=en_US.UTF-8
+LANG=en_US.UTF-8
+PYNQ_JUPYTER_NOTEBOOKS=/home/xilinx/jupyter_notebooks
+PYNQ_PYTHON=python3.10
