--- /etc/sudoers	2022-02-08 08:41:40.000000000 +0000
+++ sudoers	2022-05-05 21:55:11.808098146 +0000
@@ -6,10 +6,9 @@
 #
 # See the man page for details on how to write a sudoers file.
 #
-Defaults	env_reset
 Defaults	mail_badpass
-Defaults	secure_path="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin"
 Defaults	use_pty
+Defaults env_keep="BOARD"
 
 # This preserves proxy settings from user environments of root
 # equivalent users (group sudo)
