From 1c5dbbd272854e6e7912e2602bdfd78b64399319 Mon Sep 17 00:00:00 2001
From: David Holsgrove <david.holsgrove@xilinx.com>
Date: Mon, 22 Jul 2013 11:16:05 +1000
Subject: [PATCH 35/43] revert master-rebase changes to gdbserver

Signed-off-by: David Holsgrove <david.holsgrove@xilinx.com>
---
 gdb/gdbserver/configure.srv | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gdb/gdbserver/configure.srv b/gdb/gdbserver/configure.srv
index d19d22b3a3..7a0be5b072 100644
--- a/gdb/gdbserver/configure.srv
+++ b/gdb/gdbserver/configure.srv
@@ -210,6 +210,13 @@ case "${target}" in
 			srv_linux_usrregs=yes
 			srv_linux_thread_db=yes
 			;;
+  microblaze*-*-linux*)	srv_regobj=microblaze-linux.o
+			srv_tgtobj="linux-low.o linux-osdata.o linux-microblaze-low.o "
+			srv_tgtobj="${srv_tgtobj} linux-procfs.o linux-ptrace.o"
+			srv_linux_regsets=yes
+			srv_linux_usrregs=yes
+			srv_linux_thread_db=yes
+			;;
   powerpc*-*-linux*)	srv_regobj="powerpc-32l.o"
 			srv_regobj="${srv_regobj} powerpc-altivec32l.o"
 			srv_regobj="${srv_regobj} powerpc-cell32l.o"
-- 
2.17.1

