From 9562530bc48c76d8f824b8f4901ad90dd2969086 Mon Sep 17 00:00:00 2001
From: Nagaraju <nmekala@xilinx.com>
Date: Fri, 17 Jan 2020 15:45:48 +0530
Subject: [PATCH 43/43] Removing the header "gdb_assert.h" from MB target file

---
 gdb/microblaze-linux-tdep.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/gdb/microblaze-linux-tdep.c b/gdb/microblaze-linux-tdep.c
index e3d2a7508d..5ef937219c 100644
--- a/gdb/microblaze-linux-tdep.c
+++ b/gdb/microblaze-linux-tdep.c
@@ -37,7 +37,6 @@
 #include "linux-tdep.h"
 #include "glibc-tdep.h"
 
-#include "gdb_assert.h"
 
 #ifndef REGSET_H
 #define REGSET_H 1
-- 
2.17.1

