--- os-release.old  2022-04-18 03:28:59.000000000 -0700
+++ os-release  2022-05-05 07:43:23.604299349 -0700
@@ -1,12 +1,12 @@
-PRETTY_NAME="Ubuntu 22.04 LTS"
-NAME="Ubuntu"
-VERSION_ID="22.04"
-VERSION="22.04 (Jammy Jellyfish)"
-VERSION_CODENAME=jammy
-ID=ubuntu
-ID_LIKE=debian
-HOME_URL="https://www.ubuntu.com/"
-SUPPORT_URL="https://help.ubuntu.com/"
-BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
-PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
+NAME="PynqLinux"
+VERSION="3.0 (Belfast)"
+ID=pynqlinux
+ID_LIKE=ubuntu
+PRETTY_NAME="PynqLinux, based on Ubuntu 22.04"
+VERSION_ID="3.0"
+HOME_URL="https://www.pynq.io/"
+SUPPORT_URL="https://discuss.pynq.io/"
+BUG_REPORT_URL="https://www.pynq.io"
+PRIVACY_POLICY_URL="https://www.pynq.io"
+VERSION_CODENAME=Belfast
 UBUNTU_CODENAME=jammy
