From 13784efbd81b83bb242484f13958ee72983af023 Mon Sep 17 00:00:00 2001
From: Rock Qu <yunq@xilinx.com>
Date: Thu, 23 May 2019 11:34:58 -0700
Subject: [PATCH 1/2] irps5401

---
 drivers/hwmon/pmbus/pmbus.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/hwmon/pmbus/pmbus.c b/drivers/hwmon/pmbus/pmbus.c
index be00cd1..fe6b4eb 100644
--- a/drivers/hwmon/pmbus/pmbus.c
+++ b/drivers/hwmon/pmbus/pmbus.c
@@ -215,6 +215,7 @@ static int pmbus_probe(struct i2c_client *client,
 	{"tps544c20", 1},
 	{"tps544c25", 1},
 	{"udt020", 1},
+	{"irps5401", 5},
 	{}
 };
 
-- 
1.9.5

