--- /home/xilinx/tm_aarch64_diff/orig/etc/lsb-release	2018-08-21 16:29:46.970460260 +0000
+++ /home/xilinx/tm_aarch64_diff/final/etc/lsb-release	2018-08-21 16:13:03.814326800 +0000
@@ -1,4 +1,4 @@
-DISTRIB_ID=Ubuntu
-DISTRIB_RELEASE=20.04
-DISTRIB_CODENAME=focal
-DISTRIB_DESCRIPTION="Ubuntu 20.04 LTS"
+DISTRIB_ID=pynqlinux
+DISTRIB_RELEASE=v2.7
+DISTRIB_CODENAME=Austin
+DISTRIB_DESCRIPTION="PYNQ Linux, based on Ubuntu 20.04"

