--- a/serial-getty@.service	2022-04-07 12:28:15.000000000 -0700
+++ b/serial-getty@.service	2022-05-05 09:58:51.059920002 -0700
@@ -31,7 +31,7 @@
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --keep-baud 115200,57600,38400,9600 %I $TERM
+ExecStart=-/sbin/agetty --autologin xilinx --keep-baud 115200,57600,38400,9600 %I $TERM
 Type=idle
 Restart=always
 UtmpIdentifier=%I
