diff -ur a/gcc/config/microblaze/microblaze.c b/gcc/config/microblaze/microblaze.c
--- a/gcc/config/microblaze/microblaze.c	2017-09-19 16:06:41.169092355 +0100
+++ b/gcc/config/microblaze/microblaze.c	2017-09-28 18:02:24.846296495 +0100
@@ -2414,9 +2414,11 @@
 	{
 	  unsigned long value_long;
 	  REAL_VALUE_TYPE value;
+	  HOST_WIDE_INT value_ll;
 	  REAL_VALUE_FROM_CONST_DOUBLE (value, op);
 	  REAL_VALUE_TO_TARGET_SINGLE (value, value_long);
-	  fprintf (file, HOST_WIDE_INT_PRINT_HEX, value_long);
+	  value_ll = value_long;
+	  fprintf (file, HOST_WIDE_INT_PRINT_HEX, value_ll);
 	}
       else
 	{
@@ -2475,7 +2477,7 @@
       print_operand_address (file, XEXP (op, 0));
     }
   else if (letter == 'm')
-    fprintf (file, HOST_WIDE_INT_PRINT_DEC, (1L << INTVAL (op)));
+    fprintf (file, HOST_WIDE_INT_PRINT_DEC, (((HOST_WIDE_INT)1) << INTVAL (op)));
   else
     output_addr_const (file, op);
 }