From 254c98098a6b96d7e58d918838c49c5332e2babf Mon Sep 17 00:00:00 2001
From: hcook95 <haydencook95@gmail.com>
Date: Thu, 30 May 2019 16:18:35 -0600
Subject: [PATCH] fixed error checking conditional statement

---
 drivers/gpio/gpio-xilinx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpio/gpio-xilinx.c b/drivers/gpio/gpio-xilinx.c
index 865b0672adc6d..47dd2c5e5d2df 100644
--- a/drivers/gpio/gpio-xilinx.c
+++ b/drivers/gpio/gpio-xilinx.c
@@ -650,7 +650,7 @@ static int xgpio_of_probe(struct platform_device *pdev)
 
 	chip->clk = devm_clk_get(&pdev->dev, "s_axi_aclk");
 	if (IS_ERR(chip->clk)) {
-		if ((PTR_ERR(chip->clk) != -ENOENT) ||
+		if ((PTR_ERR(chip->clk) != -ENOENT) &&
 				(PTR_ERR(chip->clk) != -EPROBE_DEFER)) {
 			dev_err(&pdev->dev, "Input clock not found\n");
 			return PTR_ERR(chip->clk);

