From 6ef95cc45625c87b34d6e7680916b1edbb003f51 Mon Sep 17 00:00:00 2001
From: Rock Qu <yunq@xilinx.com>
Date: Fri, 30 Aug 2019 11:29:16 -0700
Subject: [PATCH] fix microblaze float

---
 gcc/config/microblaze/microblaze.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gcc/config/microblaze/microblaze.c b/gcc/config/microblaze/microblaze.c
index 9a4a287..ce72d0e 100644
--- a/gcc/config/microblaze/microblaze.c
+++ b/gcc/config/microblaze/microblaze.c
@@ -2397,9 +2397,11 @@ print_operand (FILE * file, rtx op, int letter)
       if (letter == 'F')
 	{
 	  unsigned long value_long;
+	  HOST_WIDE_INT value_ll;
 	  REAL_VALUE_TO_TARGET_SINGLE (*CONST_DOUBLE_REAL_VALUE (op),
 				       value_long);
-	  fprintf (file, HOST_WIDE_INT_PRINT_HEX, value_long);
+	  value_ll = value_long;
+	  fprintf (file, HOST_WIDE_INT_PRINT_HEX, value_ll);
 	}
       else
 	{
@@ -2458,7 +2460,7 @@ print_operand (FILE * file, rtx op, int letter)
       print_operand_address (file, XEXP (op, 0));
     }
   else if (letter == 'm')
-    fprintf (file, HOST_WIDE_INT_PRINT_DEC, (1L << INTVAL (op)));
+    fprintf (file, HOST_WIDE_INT_PRINT_DEC, (((HOST_WIDE_INT)1) << INTVAL (op)));
   else
     output_addr_const (file, op);
 }
-- 
1.9.5

