--- a/Makefile.tpl	2019-09-02 16:34:02.555314861 -0400
+++ b/Makefile.tpl	2019-09-02 16:34:36.215313964 -0400
@@ -171,7 +171,8 @@
 # built for the build system to override those in BASE_FLAGS_TO_PASS.
 EXTRA_BUILD_FLAGS = \
 	CFLAGS="$(CFLAGS_FOR_BUILD)" \
-	LDFLAGS="$(LDFLAGS_FOR_BUILD)"
+	LDFLAGS="$(LDFLAGS_FOR_BUILD)" \
+	CXXFLAGS="$(CXXFLAGS_FOR_BUILD)"
 
 # This is the list of directories to built for the host system.
 SUBDIRS = @configdirs@
--- a/Makefile.in	2019-09-02 16:34:08.427314705 -0400
+++ b/Makefile.in	2019-09-02 16:35:21.715312753 -0400
@@ -168,7 +168,8 @@
 # built for the build system to override those in BASE_FLAGS_TO_PASS.
 EXTRA_BUILD_FLAGS = \
 	CFLAGS="$(CFLAGS_FOR_BUILD)" \
-	LDFLAGS="$(LDFLAGS_FOR_BUILD)"
+	LDFLAGS="$(LDFLAGS_FOR_BUILD)" \
+	CXXFLAGS="$(CXXFLAGS_FOR_BUILD)"
 
 # This is the list of directories to built for the host system.
 SUBDIRS = @configdirs@
