--- etc/environment
+++ etc/environment
@@ -1 +1,5 @@
-PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
+PATH="/opt/microblazeel-xilinx-elf/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"
+LC_ALL=en_US.UTF-8
+LANG=en_US.UTF-8
+PYNQ_JUPYTER_NOTEBOOKS=/home/xilinx/jupyter_notebooks
+PYNQ_PYTHON=python3
