--- etc/legal
+++ etc/legal
@@ -1,8 +1,8 @@
 
-The programs included with the Ubuntu system are free software;
+The programs included with the PYNQ Linux system are free software;
 the exact distribution terms for each program are described in the
 individual files in /usr/share/doc/*/copyright.
 
-Ubuntu comes with ABSOLUTELY NO WARRANTY, to the extent permitted by
+PYNQ Linux comes with ABSOLUTELY NO WARRANTY, to the extent permitted by
 applicable law.
 
