--- etc/sudoers
+++ etc/sudoers
@@ -6,9 +6,8 @@
 #
 # See the man page for details on how to write a sudoers file.
 #
-Defaults	env_reset
 Defaults	mail_badpass
-Defaults	secure_path="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin"
+Defaults env_keep="BOARD"
 
 # This fixes CVE-2005-4890 and possibly breaks some versions of kdesu
 # (#1011624, https://bugs.kde.org/show_bug.cgi?id=452532)
