root@xsjengvm209187:/etc/update-motd.d# more 10-help-text.diff
--- 10-help-text        2021-05-20 16:46:59.000000000 +0000
+++ 10-help-text.new    2021-05-20 16:47:27.000000000 +0000
@@ -20,7 +20,4 @@
 #    with this program; if not, write to the Free Software Foundation, Inc.,
 #    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

-printf "\n"
-printf " * Documentation:  https://help.ubuntu.com\n"
-printf " * Management:     https://landscape.canonical.com\n"
-printf " * Support:        https://ubuntu.com/advantage\n"
+
